apps/web/app/queries/protocols/use-protocol-fee.ts

```diff
@@ -0,0 +1,37 @@
+import { useMemo } from 'react';
+
+import { useQuery } from '@tanstack/react-query';
+import { ProtocolSlug } from '~/features/stacking/start-liquid-stacking/utils/types-preset-protocols';
+import { getNetworkInstanceByName } from '~/features/stacking/start-pooled-stacking/utils/utils-stacking-pools';
+import { createEmptyQueryOptions } from '~/queries/create-empty-query-options';
+import { createGetDAOFeeQueryOptions } from '~/queries/protocols/dao/fee';
+import { CreateProtocolFeeQueryOptionsParams } from '~/queries/protocols/protocol-types';
+import { useStacksClient } from '~/queries/stacks/stacks-client';
+import { useLeatherConnect } from '~/store/addresses';
+import { useStacksNetwork } from '~/store/stacks-network';
+
+export function useProtocolFee(slug: ProtocolSlug) {
+  const client = useStacksClient();
+  const { stacksAccount } = useLeatherConnect();
+  const address = stacksAccount?.address;
+
+  const network = useStacksNetwork();
+  const networkMode = getNetworkInstanceByName(network.networkName);
+
+  const queryOptions = useMemo(() => {
+    const options: CreateProtocolFeeQueryOptionsParams = {
+      address,
+      client,
+      networkMode,
+    };
+
+    switch (slug) {
+      case 'stacking-dao':
+        return createGetDAOFeeQueryOptions(options);
+      default:
+        return createEmptyQueryOptions();
+    }
+  }, [address, client, networkMode, slug]);
```

Comment:
I do wonder if a better approach is to just have each their own route and coming separately rather than having conditional logic everywhere in the code?

Reply:
sep routes? I use it in table to display fees
I will need to add conditionals all the same in the table, with different queries

Reply:
all good for now, but later I think we should separate in routes more, even if this specifically is used in table